src/hypothesis/strategies.py

```diff
@@ -910,6 +912,108 @@ def do_draw(self, data):
+        tz-aware datetimes from this strategy may be ambiguous or non-existent
+        due to daylight savings, leap seconds, timezone and calendar
+        adjustments, etc.  This is intentional, as malformed timestamps are a
+        common source of bugs.  Consider validating all datetime values, or
+        working purely in UTC to avoid such issues.
+
+    :py:func:`hypothesis.extra.timezones` requires the ``pytz`` package, but
+    provides all timezones in the Olsen database.  If you also want to allow
+    naive datetimes, combine strategies like ``none() | timezones()``.
+
+    Alternatively, you can create a list of the timezones you wish to allow
+    (e.g. from the standard library, ``datetutil``, or ``pytz``) and use
+    :py:func:`sampled_from`.  Ensure that simple values such as None or UTC
+    are at the beginning of the list for proper minimisation.
+
+    """
+    from hypothesis.searchstrategy.datetime import DatetimeStrategy
+
+    check_type(dt.datetime, min_datetime, 'min_datetime')
+    check_type(dt.datetime, max_datetime, 'max_datetime')
+    if min_datetime.tzinfo is not None:
+        raise InvalidArgument('min_datetime=%r must not have tzinfo'
+                              % min_datetime)
+    if max_datetime.tzinfo is not None:
+        raise InvalidArgument('max_datetime=%r must not have tzinfo'
+                              % max_datetime)
+    check_valid_interval(min_datetime, max_datetime,
+                         'min_datetime', 'max_datetime')
+    if timezones is None:
+        if min_datetime == max_datetime:
+            return just(min_datetime)
+        timezones = none()
+    if not isinstance(timezones, SearchStrategy):
+        raise InvalidArgument(
+            'timezones=%r must be a SearchStrategy that can provide tzinfo '
+            'for datetimes (either None or dt.tzinfo objects)' % (timezones,))
+    return DatetimeStrategy(min_datetime, max_datetime, timezones)
+
+
+@defines_strategy
+def dates(min_date=dt.date.min, max_date=dt.date.max):
+    """A strategy for dates between ``min_date`` and ``max_date``."""
+    check_type(dt.date, min_date, 'min_date')
+    check_type(dt.date, max_date, 'max_date')
+    check_valid_interval(min_date, max_date, 'min_date', 'max_date')
+    if min_date == max_date:
+        return just(min_date)
+    return datetimes(min_datetime=dt.datetime.combine(min_date, dt.time.min),
+                     max_datetime=dt.datetime.combine(max_date, dt.time.max)
+                     ).map(dt.datetime.date)
+
+
+@defines_strategy
+def times(min_time=dt.time.min, max_time=dt.time.max, timezones=None):
+    """A strategy for times between ``min_time`` and ``max_time``.
+
+    The ``timezones`` argument is handled as for :py:func:`datetimes`.
+
+    """
+    check_type(dt.time, min_time, 'min_time')
+    check_type(dt.time, max_time, 'max_time')
+    if min_time.tzinfo is not None:
+        raise InvalidArgument('min_time=%r must not have tzinfo' % min_time)
```

Comment:
That both must be naive is the second line of the docstring.  The reasoning as to *why* is I think an implementation detail, but I'll add a comment so anyone reading the source has something to go on.